crowdfunding/projects/views.py

```diff
@@ -1,3 +1,121 @@
 from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from django.http import Http404
+from rest_framework import status, generics, permissions
+from django_filters.rest_framework import DjangoFilterBackend
 
-# Create your views here.
+from .models import Project, Pledge
+from .serializers import ProjectSerializer, PledgeSerializer, ProjectDetailSerializer
+from .permissions import IsOwnerOrReadOnly, IsSupporterOrReadOnly
+
+class ProjectList(APIView):
+
+	permission_classes = [permissions.IsAuthenticatedOrReadOnly,
+	IsOwnerOrReadOnly]
+
+	def get(self, request):
+		projects = Project.objects.all()
+		serializer = ProjectSerializer(projects, many=True)
+		return Response(serializer.data)
+
+	def post(self, request):
+		serializer = ProjectSerializer(data=request.data)
+		if serializer.is_valid():
+			serializer.save(owner=request.user)
+			return Response(serializer.data, status=status.HTTP_201_CREATED)
+		return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class ProjectListFilter(generics.ListAPIView):
+	queryset = Project.objects.all()
+	serializer_class = ProjectSerializer
+	filter_backends = [DjangoFilterBackend]
+	filterset_fields = ['owner', 'date_created', 'is_open']
```

Comment:
you could probably have replaced the ProjectList class with this one, if you added the permissions classes and made it a listcreateAPIView